agent/discovery/discovery.go

```diff
@@ -24,9 +24,9 @@ func (d BaseDiscovery) GetId() string {
 }
 
 // Execute one iteration of a discovery and store the result in the Consul KVStore.
-func (d BaseDiscovery) Discover() error {
+func (d BaseDiscovery) Discover() (string, error) {
 	d.host, _ = os.Hostname()
-	return nil
+	return "", nil
```

Comment:
We could put something like `Basic discovery example`

Reply:
Changed!